### PR TITLE
feat(annotations): add redux

### PR DESCRIPTION
**Pull request by hoorayimhelping**
Connects #376

Adds the first function action creator, thunk, and reducer to the annotations redux store.

- Builds on the work of Jill and Slotman
- Lays the groundwork for future redux stuff, probably
- About that laying groundwork: adds a reducer test

here's a picture of redux tools showing the reducer being called via a call to the thunk that was added in `GetResource`
Screen Shot 2021-01-26 at 4 00 12 PM

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

File: src/annotations/api/index.ts [hunk 1]
```diff
@@ -41,9 +41,24 @@ export const writeAnnotation = async (
...
+
+  if (res.status >= 300) {
+    throw new Error(res.data?.message)
+  }
+
+  return res.data.map((retrievedAnnotation: AnnotationStream) => ({
+    stream: retrievedAnnotation.stream,
+    annotations: retrievedAnnotation.annotations,
+  }))
+}
```
Comment (hoorayimhelping) [inline src/annotations/api/index.ts:57]: we'll want to figure out the proper API here vis-a-vis `getAnnotation` and `getAnnotations`

**Comment by hercules-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 50 | :red_circle:
accessibility | 65 | :yellow_circle:
best-practices | 87 | :green_circle:
seo | 80 | :yellow_circle:
pwa | null | :question:

File: src/annotations/actions/thunks.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import {getAnnotations} from 'src/annotations/api'
+import {Dispatch} from 'react'
+
+import {
+  setAnnotations,
+  Action as AnnotationAction,
+} from 'src/annotations/actions/creators'
+
+export const fetchAndSetAnnotations = () => async (
```
Comment (ShmuelLotman) [inline src/annotations/actions/thunks.ts:9]: Love the naming here. I dislike it when fetch operations have a fetch<thing> call and then a set<thing> action... gets confusing to look at before coffee.

**Review by ShmuelLotman — APPROVED** [review]
Approving. This looks good to me. Thanks for bringing this across the finish line!

File: src/annotations/reducers/index.test.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import {annotationsReducer, initialState} from 'src/annotations/reducers'
+
+const mockAnnotations = [
+  {
+    stream: 'default',
+    annotations: [
+      {
+        summary: 'hi',
+        end: '2021-01-21T00:05:51Z',
```
Comment (jrenee42) [inline src/annotations/reducers/index.test.ts:9]: note that the 'end' and 'start' are strings.

the plot is expecting numbers (milliseconds) not date strings.
unless we want to store them like that and do conversions.

besides that the redux looks good.

Comment (hoorayimhelping) [reply src/annotations/reducers/index.test.ts:9]: created an issue to address this: #570

**Pull request merged**